ZuluContent/Items/Containers/Bridle.cs

```diff
@@ -125,13 +144,25 @@ public override void Serialize(IGenericWriter writer)
             base.Serialize(writer);
 
             writer.Write((int) 0); // version
+
+            ICraftable.Serialize(writer, this);
+
+            writer.WriteEncodedInt((int) Mark);
```

Comment:
Sorry might not have explained it well before, but you can push this line:
```csharp
writer.WriteEncodedInt((int) Mark);
```

 into `ICraftable.Serialize` saving you the hassle of doing it in every class that inherits the interface. e.g.

```csharp
public static void Serialize(IGenericWriter writer, ICraftable item)
{
    writer.Write(Version);
    writer.WriteEncodedInt((int) Mark); 
    writer.Write(item.PlayerConstructed);

    writer.Write(item.Crafter != null);
    if (item.Crafter != null)
    {
        writer.Write(item.Crafter);
    }
}

public static void Deserialize(IGenericReader reader, ICraftable item)
{
    var version = reader.ReadInt();
    Mark = (MarkQuality) reader.ReadEncodedInt();
    item.PlayerConstructed = reader.ReadBool();

    if (reader.ReadBool())
        item.Crafter = reader.ReadEntity<Mobile>();
}
```

You can potentially go this route for `IResource` as well if you wanted to, but its just one property at the moment so maybe not worth it unless we start to serialize more.

Reply:
Yeah maybe not worth it will leave it as is for now